### ISSUE TITLE
RS2008 false-negative with target-typed 'new' expression

### ISSUE THREAD

**Issue by david-acker** [dotnet/roslyn-analyzers#6393]
### Analyzer

**Diagnostic ID**: [RS2008](https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.md#rs2008-enable-analyzer-release-tracking): `Enable analyzer release tracking`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.Analyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.Analyzers)

**Version**: 3.3.3 (transitive dependency of [Microsoft.CodeAnalysis.CSharp](https://www.nuget.org/packages/Microsoft.CodeAnalysis.CSharp) Version 4.4.0 )

### Describe the bug

The [DiagnosticDescriptionCreationAnalyzer](https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs) does not report RS2008 if a target-typed 'new' expression is used to instantiate the `DiagnosticDescriptor`.

### Steps to Reproduce

1. Create a new C# project in Visual Studio.
2. Install the Microsoft.CodeAnalysis.CSharp NuGet package.
3. Copy the following code sample:
```csharp
using Microsoft.CodeAnalysis;

public class Test
{
    // Correctly reports RS2008 on the DiagnosticDescriptor "id" parameter 
    public static readonly DiagnosticDescriptor DiagnosticWithExplicitType = new DiagnosticDescriptor(
        "TEST0001",
        "Title",
        "Message",
        "Category",
        DiagnosticSeverity.Error,
        isEnabledByDefault: true);

    // Does not report RS2008
    public static readonly DiagnosticDescriptor DiagnosticWithTargetTypedExpression = new(
        "TEST0002",
        "Title",
        "Message",
        "Category",
        DiagnosticSeverity.Error,
        isEnabledByDefault: true);
}
```

### Expected behavior
The analyzer should report RS2008 when the `DiagnosticDescriptor` is instantiated using a target-typed expression.

### Actual behavior
The analyzer does not report the diagnostic.

#### Note
This issue only seems to occur when the analyzer is run by Visual Studio. Updating the tests [ReleaseTrackingAnalyzerTests](https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/ReleaseTrackingAnalyzerTests.cs) to use a target-typed 'new' expression doesn't cause any test failures related to this issue.

**Comment by Youssef1313** [comment]
This was fixed in #6119

Can you try with newer versions? e.g, [3.3.4-beta1.22559.1](https://www.nuget.org/packages/Microsoft.CodeAnalysis.Analyzers/3.3.4-beta1.22559.1)

**Comment by david-acker** [comment]
Ah that explains why unit tests passed then.

I updated Microsoft.CodeAnalysis.Analyzers to the most recent prerelease version and am not seeing the issue anymore. Thanks!
